## Exploratory Data Analysis (EDA)

EDA (Exploratory Data Analysis) is the process of understanding a dataset before applying machine learning. It helps us identify the data structure, missing values, data types, and basic statistics.

## Import Libraries

Import the required libraries for data analysis.

In [2]:
import pandas as pd #use for data manipulation and analysis
import numpy as np #use for numerical computations

## Load Dataset

Load the Titanic dataset into a pandas DataFrame.

In [3]:
df = pd.read_csv("train.csv")

## Preview Dataset

Display the first five rows of the dataset.

In [4]:
df.head() #to see the first 5 rows of the dataset

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


## Dataset Information

View the dataset structure, data types, and missing values.

In [5]:
df.info() #to see the information about the dataset

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


## Statistical Summary

Generate summary statistics for numerical columns.

In [6]:
df.describe() #to see the statistical information about the dataset

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


## Dataset Shape

Check the number of rows and columns.

In [7]:
df.shape #no of rows and columns in the dataset

(891, 12)

## Column Names

Display all column names.

In [8]:
df.columns # to see the column names of the dataset

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='str')

## Data Types

Identify numerical and categorical columns.

In [9]:
df.dtypes #to see the data types of the columns in the dataset

PassengerId      int64
Survived         int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object

In [14]:
numerical=df.select_dtypes(include=['int64','float64']).columns #to see the columns with numerical data types
categorical=df.select_dtypes(include=['object']).columns #to see the columns with categorical data types

print("Numerical Columns: ",numerical)
print("Categorical Columns: ",categorical)

Numerical Columns:  Index(['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare'], dtype='str')
Categorical Columns:  Index(['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked'], dtype='str')


C:\Users\DELL\AppData\Local\Temp\ipykernel_20508\260075319.py:2: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical=df.select_dtypes(include=['object']).columns #to see the columns with categorical data types


## Missing Values

Count missing values in each column.

In [15]:
df.isnull().sum() #to see the number of missing values in each column

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

## Data Story

The Titanic dataset contains passenger information and their survival status. It has 891 rows and 12 columns. Some columns, such as Age and Cabin, contain missing values. The dataset includes both numerical and categorical features. This analysis provides a basic understanding of the data before preprocessing and machine learning.